api/src/main/java/uk/gov/dhsc/htbhf/claimant/entitlement/AdditionalPregnancyVoucherCalculator.java

```diff
@@ -0,0 +1,75 @@
+package uk.gov.dhsc.htbhf.claimant.entitlement;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.dhsc.htbhf.claimant.entity.PaymentCycle;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Calculates the number of additional pregnancy vouchers when a claim has been updated to include an expected delivery date.
+ */
+@Component
+public class AdditionalPregnancyVoucherCalculator {
+
+    private final Integer entitlementCalculationDurationInDays;
+    private final PregnancyEntitlementCalculator pregnancyEntitlementCalculator;
+    private final int vouchersPerPregnancy;
+
+    public AdditionalPregnancyVoucherCalculator(@Value("${entitlement.number-of-vouchers-per-pregnancy}") int vouchersPerPregnancy,
+                                                PaymentCycleConfig paymentCycleConfig,
+                                                PregnancyEntitlementCalculator pregnancyEntitlementCalculator) {
+        this.entitlementCalculationDurationInDays = paymentCycleConfig.getEntitlementCalculationDurationInDays();
+        this.pregnancyEntitlementCalculator = pregnancyEntitlementCalculator;
+        this.vouchersPerPregnancy = vouchersPerPregnancy;
+    }
+
+    /**
+     * Get the number of additional pregnancy vouchers when we've been notified that a claimant is now pregnant.
+     * The pregnancy entitlement calculator is called for each full week that is left of the current cycle.
+     * e.g if the claim was updated mid way through week two of the cycle, the user will get pregnancy vouchers for week 3 and 4 (given a four week cycle).
+     * if the claim was updated during week four of the cycle, no additional vouchers will be returned.
+     *
+     * @param expectedDueDate  the expected due date of the claimant
+     * @param paymentCycle     the current payment cycle
+     * @param claimUpdatedDate the date that the claim was updated with the new expected due date
+     * @return the number of additional pregnancy vouchers the claimant is entitled to
+     */
+    public int getAdditionalPregnancyVouchers(LocalDate expectedDueDate,
+                                              PaymentCycle paymentCycle,
+                                              LocalDate claimUpdatedDate) {
+        if (!claimUpdatedDate.isAfter(paymentCycle.getCycleStartDate())) {
+            // if the update happened on the same day as the payment cycle start date,
```

Comment:
Is there the possibility that if the request is on the same date as the cycle start date that it might happen after we've already made the calculation and payment, so might have missed the payment and not get paid? Could be a timing issue here.

Reply:
@YetAnotherMatt Did we discuss this scenario?

Reply:
We discussed what would happen if there was no current paymentCycle. I think Mike is on to something here.

Reply:
Updated with additional check to make sure the payment cycle has also not been processed already.